### PR TITLE
Fix CHECK_BSD_STYLE_MACROS for OpenBSD and cryptodev-linux

### PR DESCRIPTION
**Pull request by Retropotenza**
Fixes https://github.com/openssl/openssl/issues/8911

CLA: trivial

This fix works on:

- FreeBSD, where both CRYPTO_ALGORITHM_MIN and CRYPTO_ALGORITHM_MAX are known to the preprocessor,
- OpenBSD, where CRYPTO_ALGORITHM_MIN will be automatically set by the preprocessor to 0 (lower than the defined CRYPTO_ALGORITHM_MAX),
- cryptodev-linux, where the preprocessor will set CRYPTO_ALGORITHM_MIN to 0 and CRYPTO_ALGORITHM_MAX to (0 - 1).

References:
- https://github.com/freebsd/freebsd/blob/master/sys/opencrypto/cryptodev.h#L167-L209
- https://github.com/openbsd/src/blob/master/sys/crypto/cryptodev.h#L108
- https://github.com/cryptodev-linux/cryptodev-linux/blob/f971e0cd4a0ebe59fb2e8e17240399bf6901b09b/crypto/cryptodev.h#L57

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by levitte** [comment]
In the commit message, I would suggest changing "Proposed fix for" to simply "Fixes".

**Comment by slontis** [comment]
Hi @Retropotenza
CLA: trivial needs to be added as a seperate line to the actual 'commit' message.
>git commit --amend

**Comment by Retropotenza** [comment]
Nope... `git commit --amend -m "..."` didn't seem to work unfortunately. Sorry for the multiple pushes here, at the end I remove one space to make a real commit. I hope that the continuous integration scripts will be successful now (they where before...).

File: engines/e_devcrypto.c [hunk 1]
```diff
@@ -25,8 +25,8 @@
 
 /* #define ENGINE_DEVCRYPTO_DEBUG */
 
-#ifdef CRYPTO_ALGORITHM_MIN
-# define CHECK_BSD_STYLE_MACROS
+#if CRYPTO_ALGORITHM_MIN < CRYPTO_ALGORITHM_MAX
+#define CHECK_BSD_STYLE_MACROS
```
Comment (levitte) [inline engines/e_devcrypto.c]: This should be indented by adding a space between `#` and `define`

**Comment by t8m** [comment]
Pushed c596b8d

**Comment by t8m** [comment]
1.1.1 branch
3268087d2cca5f5a882c9620849a12915a4ab0f1

**Pull request closed without merging**